### PR TITLE
Fixed #26710 -- Made CreateModel.references_model take app_label into account.

### PR DESCRIPTION
**Pull request by sir-sigurd**
https://code.djangoproject.com/ticket/26710

### CODE REVIEW

File: django/db/migrations/operations/models.py [hunk 1]
```diff
@@ -104,20 +104,25 @@ def describe(self):
         return "Create %smodel %s" % ("proxy " if self.options.get("proxy", False) else "", self.name)
 
     def references_model(self, name, app_label=None):
-        strings_to_check = [self.name]
+        name_lower = name.lower()
+        if name_lower == self.name_lower:
```
Comment (charettes) [inline django/db/migrations/operations/models.py]: Shouldn't this also take into account `app_label`?

File: django/db/migrations/operations/models.py [hunk 2]
```diff
@@ -104,20 +104,25 @@ def describe(self):
...
+            return True
+
+        models_to_check = []
```
Comment (charettes) [inline django/db/migrations/operations/models.py]: Initialize `model_to_check` with `list(self.bases)`.

File: django/db/migrations/operations/models.py (hunk 1 above)
Comment (sir-sigurd) [inline django/db/migrations/operations/models.py]: It seems that this couldn't be done with the current approach.
I think that we could add `current_app` argument to `references_model` and `references_field` to handle it, but probable this should go to another PR.

Comment (charettes) [inline django/db/migrations/operations/models.py]: Does `if app_label is None and name_lower == self.name_lower` breaks any tests?

Comment (sir-sigurd) [inline django/db/migrations/operations/models.py]: Yes.

Comment (charettes) [inline django/db/migrations/operations/models.py]: Alright, let's keep this change for another PR then.

**Comment by timgraham** [comment]
merged in 7fc4c1db627807874966e4f96b34cff7c9af886f, thanks!

**Comment by millin** [comment]
@timgraham, could you apply it to stable?

**Comment by timgraham** [comment]
If it's not a regression from a previous release or a major bug in a new feature it won't be backported. See our [supported versions policy](https://docs.djangoproject.com/en/dev/internals/release-process/#supported-versions).
